### PR TITLE
Refactor documentation of records.

### PR DESCRIPTION
**Pull request by Zimmi48**
This fixes bug https://coq.inria.fr/bugs/show_bug.cgi?id=4971

### CODE REVIEW

**Comment by herbelin** [comment]
Not that I know how to do it easily, or even automatically, but would you mind attaching a link to an html or pdf to show the new output which would be easier to read than LaTeX?

**Comment by Zimmi48** [comment]
@SkySkimmer Any chance that we can use GitLab artifacts for such things?

**Comment by Zimmi48** [comment]
@herbelin is there a way to build only a chapter of the manual in PDF? The full manual is too big to be stored as an attachment on Bugzilla.

**Comment by SkySkimmer** [comment]
Gitlab is already set up to produce documentation artifacts, so you just need to push the branch and wait for completion. See eg https://gitlab.com/Zimmi48/coq/builds/17834300/artifacts/download
I'm not familiar with Coq's documentation system so there might be some issues (eg a missing font causing bad characters or what have you), please report if that's the case.

**Comment by herbelin** [comment]
Hi, don't want to make things complicated for PR submitters... To answer the question though, `pdftk` can be used to extract pages from a pdf. Following "artifacts" links like @SkySkimmer proposed seems otherwise also ok.

**Comment by Zimmi48** [comment]
I've pushed a new version including the relevant parts of 0162292. This should make the merge in trunk simpler.

**Comment by Zimmi48** [comment]
Note I've also a branch cherry-picking this commit on trunk: https://github.com/Zimmi48/coq/tree/trunk%2Bcherry-pick-refactor-records-doc
This branch is currently building. When the build is over, the manual should normally be available [here](https://gitlab.com/Zimmi48/coq/builds/18224648/artifacts/file/install/share/doc/coq/pdf/Reference-Manual.pdf).
Thanks to @SkySkimmer for the info! I had to do the cherry-picking on trunk though because `.gitlab-ci.yml` isn't in v8.6.

**Comment by maximedenes** [comment]
@Zimmi48 What is the status of this PR? Can I merge it?

**Comment by Zimmi48** [comment]
The link to the PDF above has been updated. I'd appreciate to have @herbelin or @SkySkimmer review the result.

File: doc/refman/RefMan-ext.tex [hunk 1]
```diff
@@ -40,19 +40,19 @@ \section{Record types
...
 
 \smallskip
-{\tt Record} {\ident} {\params} \texttt{:} 
-   {\sort} := {\ident$_0$} \verb+{+
- {\ident$_1$} \binders$_1$ \texttt{:} {\term$_1$}; 
-              \dots
-  {\ident$_n$} \binders$_n$ \texttt{:} {\term$_n$} \verb+}+.
+{\tt Record {\ident} {\params} : {\sort} := {\ident$_0$} \{
+  {\ident$_1$} \binders$_1$ : {\term$_1$} ; ... ;
+  {\ident$_n$} \binders$_n$ : {\term$_n$} \}.}
```
Comment (SkySkimmer) [inline doc/refman/RefMan-ext.tex]: This part could use some newlines in the output.

File: doc/refman/RefMan-ext.tex [hunk 2]
```diff
@@ -139,33 +110,119 @@ \section{Record types
...
+\item Records defined with the {\tt Record} keyword are not allowed to be
+recursive (references to the record's name in the type of its field
+raises an  error). To define recursive records, one can use the {\tt
+Inductive} and {\tt CoInductive} keywords, resulting in an inductive
+or co-inductive record.
+A \emph{caveat}, however, is that records
 cannot appear in mutually inductive (or co-inductive) definitions.
-Induction schemes are automatically generated for inductive records.
+
+\item Induction schemes are automatically generated for inductive records.
```
Comment (SkySkimmer) [inline doc/refman/RefMan-ext.tex]: This part is a bit vague. Is an "inductive record" just a record like thing defined with "Inductive"?

**Comment by SkySkimmer** [comment]
LGTM

File: doc/refman/RefMan-ext.tex (hunk 2 above)
Comment (Zimmi48) [inline doc/refman/RefMan-ext.tex]: The remark just above says "resulting in an inductive or co-inductive record", so yes.

File: doc/refman/RefMan-ext.tex (hunk 1 above)
Comment (Zimmi48) [inline doc/refman/RefMan-ext.tex]: Agreed and updated.

**Comment by herbelin** [comment]
I had read once and I was ok. I'm afraid I'm now missing the ability of a fresh look to give relevant comments.

**Comment by Zimmi48** [comment]
That's fine I think this is ready ;-)

**Pull request merged**